### PR TITLE
feat: refined clear commands

### PR DESCRIPTION
**Pull request by savedra1**

### CODE REVIEW

File: config/history.go [hunk 1]
```diff
@@ -220,35 +248,26 @@ func AddClipboardItem(text, fp string) error {
...
+	if err := WriteUpdate(data); err != nil {
 		return err
 	}
 	return nil
 }
 
 // This pins and unpins an item in the clipboard
 func TogglePinClipboardItem(timeStamp string) (bool, error) {
-	var data ClipboardHistory
+	var data ClipboardHistory = fileContents()
```
Comment (ccoVeille) [inline config/history.go:268]: I would say choose your team


```suggestion
	 data := fileContents()
```

Or


```suggestion
	var data ClipboardHistory
data = fileContents()
```

File: config/history.go [hunk 2]
```diff
@@ -220,35 +248,26 @@ func AddClipboardItem(text, fp string) error {
...
-	if err != nil {
-		return pinned, err
-	}
-
-	err = json.Unmarshal(fileData, &data)
-	if err != nil {
-		return pinned, err
-	}
-
 	for i := range data.ClipboardHistory {
```
Comment (ccoVeille) [inline config/history.go:271]: Wait

Why this?

```
         for i := range data.ClipboardHistory { 
                  if data.ClipboardHistory[i].Recorded == timeStamp {
```

And not this ?

```suggestion
 	for _, ch := range data.ClipboardHistory {
       if ch.Recorded == timeStamp
```

Comment (savedra1) [reply config/history.go:271]: Also agree. Will get this and the above updated.

File: config/history.go [hunk 3]
```diff
@@ -80,6 +80,20 @@ func DisplayServer() string {
...
 
+func Paths() (string, string) {
+	/* Returns full path string for clipboard file.
+	useful when needing to be accessed form a
+	bubbletea method.
+	*/
+	currentUser, err := user.Current()
+	utils.HandleError(err)
+	// Construct the path to the config directory
+	clipseDir := filepath.Join(currentUser.HomeDir, ".config", clipseDir)
```
[the remark this answers is not in the document]
Comment (savedra1) [reply config/history.go:91]: @ccoVeille I appreciate you taking the time here. I will have a look into this.

File: config/history.go (hunk 2 above)
Comment (savedra1) [reply config/history.go:271]: Oh, I just remembered the reason for this.. it saves you from having to re-pack an array with new values to then write to the history file. Eg, the alternative would mean something like:
```go
var updatedClipboardHistory []ClipboardItem
	for _, item := range data.ClipboardHistory {
		if item.Recorded == timeStamp {
			pinned = item.Pinned
			item.Pinned = !item.Pinned
		}
		updatedClipboardHistory = append(updatedClipboardHistory, item)
	}
``` 
This doesn't feel better, but maybe I'm missing something?

Comment (ccoVeille) [reply config/history.go:271]: OK, indeed

I missed you reassigned it

Then it should be something like this (pseudo code)

```
var updatedClipboardHistory []ClipboardItem

```go
for i, item := range data.ClipboardHistory { 	
	if item.Recorded == timeStamp {
      pinned = item.Pinned
      item.Pinned = !item.Pinned
      data.ClipboardHistory[i] = item
      break
}
```

Comment (ccoVeille) [reply config/history.go:271]: I edited my previous comment if you saw my first version

Comment (savedra1) [reply config/history.go:271]: @ccoVeille Yess thanks I like that a lot more. Have been a bit hasty merging things but but create a new pr over [here](https://github.com/savedra1/clipse/pull/73/commits/a08ba2d7f5704509c41e212ea877f7e31b23b8cd)

**Pull request merged**